### PR TITLE
Made mapping between `ResampleAlg` and `GDALRIOResampleAlg` more direct.

### PR DESCRIPTION
**Pull request by metasim**
Before the private function `map_resample_alg` was the only way to get the expected GDAL enumeration value.

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Before this change, you had to copy `map_resample_alg` (because it was private) when calling into `gdal-sys` functions needing the resampling algorithm (e.g. `GDALAutoCreateWarpedVRT`). Instead of making `pub fn map_resample_alg ...`, took the approach that more tightly binds the C-side types to the Rust enum. Makes the association more clear, and less error prone in the future.

### CODE REVIEW

File: src/raster/rasterband.rs [hunk 1]
```diff
@@ -17,37 +17,49 @@ use ndarray::Array2;
...
+    Average = GDALRIOResampleAlg::GRIORA_Average,
     /// Mode (selects the value which appears most often of all the sampled points)
-    Mode,
+    Mode = GDALRIOResampleAlg::GRIORA_Mode,
     /// Gauss blurring
-    Gauss,
+    Gauss = GDALRIOResampleAlg::GRIORA_Gauss,
 }
 
-fn map_resample_alg(alg: &ResampleAlg) -> u32 {
```
Comment (metasim) [inline src/raster/rasterband.rs:41]: Note: this was not private before, so there was no easy way to map values when calling into lower level `gdal-sys` functions.

File: src/raster/rasterband.rs [hunk 2]
```diff
@@ -17,37 +17,49 @@ use ndarray::Array2;
...
+/// let stats = band1.get_statistics(true, false)?.unwrap();
+/// // Down-sample a image using cubic-spline interpolation
+/// let buf = band1.read_as::<f64>((0, 0), ds.raster_size(), (2, 2), Some(ResampleAlg::CubicSpline))?;
+/// // In this particular image, resulting data should be close to the overall average.
+/// assert!(buf.data.iter().all(|c| (c - stats.mean).abs() < stats.std_dev / 2.0));
+/// # Ok(())
+/// # }
+/// ```
 #[derive(Debug, Copy, Clone)]
+#[repr(u32)]
```
Comment (metasim) [inline src/raster/rasterband.rs:39]: Not entirely sure about this value, or even if it should be specified.
`GDALRIOResampleAlg::Type` is `libc::c_uint`, which is `u32` on *nix. The [repr options](https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#repr-annotations-accepted-on-enums) maybe suggest `#[repr(C)]` should be used instead, and the right value will be selected based on the discriminant values? Advice appreicated.

File: src/raster/rasterband.rs (hunk 1 above)
Comment (metasim) [reply src/raster/rasterband.rs:41]: Also, the return type should have been `GDALRIOResampleAlg::Type`.

File: src/raster/rasterband.rs [hunk 3]
```diff
@@ -17,37 +17,49 @@ use ndarray::Array2;
...
+/// let ds = Dataset::open("fixtures/tinymarble.tif")?;
+/// let band1 = ds.rasterband(1)?;
+/// let stats = band1.get_statistics(true, false)?.unwrap();
+/// // Down-sample a image using cubic-spline interpolation
+/// let buf = band1.read_as::<f64>((0, 0), ds.raster_size(), (2, 2), Some(ResampleAlg::CubicSpline))?;
+/// // In this particular image, resulting data should be close to the overall average.
+/// assert!(buf.data.iter().all(|c| (c - stats.mean).abs() < stats.std_dev / 2.0));
+/// # Ok(())
+/// # }
+/// ```
```
Comment (metasim) [inline src/raster/rasterband.rs:37]: Bonus documentation while I was in there :)

File: src/raster/rasterband.rs (hunk 2 above)
Comment (lnicola) [reply src/raster/rasterband.rs:39]: I don't think we need the `#[repr]` because of the cast in `to_gdal`.

**Review by lnicola — APPROVED** [review]

**Pull request merged**